Documentation/Design/.NET Tool v2.md

```diff
@@ -0,0 +1,71 @@
+# Functional design .NET tool 2.0 `coverlet`
+
+Tracking issue https://github.com/tonerdo/coverlet/issues/683
+
+The idea is to support some new verbs:
+
+# coverlet run
+
+This will support current instrumentation workflow
+
+```
+coverlet run /path/to/test-assembly.dll --target "dotnet" --targetargs "test /path/to/test-project --no-build"
+```
+
+# coverlet mergereports
+
+Merge `json` default coverlet reports because we should otherwise implement merge for every supported format a huge effort for first version and also with the hope to integrate this feature directly inside collectors.
+
+We'll add an alias to `json` format called `coverlet` to distiguish from possible other future json format and also because this report make sense only inside specific test lifecycle.
+The content of this report is releated to internal structure and was not born to be "portable" or "stable" over time.
+
+```
+coverlet mergereports --coverletreport .\**\coverlet.json --targetdir .\Report\coverlet.xml --format cobertura
```

Comment:
I think it's better that the filenames are command line arguments, rather than having to specify the `--coverletreport` flag. Also rely on bash filename expansion primarily, but perhaps have support for `**` globbing in the tool.

Reply:
> I think it's better that the filenames are command line arguments,

Can you explain better, it's not clear to me....can you type a command sample?

Reply:
coverlet mergereports --targetdir .\Report\coverlet.xml --format cobertura .\**\coverlet.json previously-merged.json ...